Sources/Nimble/Matchers/BeResult.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+///
+/// A Nimble matcher for Result that succeeds when the actual value is success
+///
+/// You can pass a closure to do any arbitrary custom matching
+/// to the value inside result. The closure only gets called when result is success.
+#if swift(>=5)
+@available(swift 5)
+public func beSuccess<T>(test: ((T) -> Void)? = nil) -> Predicate<Result<T, Error>> {
```

Comment:
This is awesome and I want it! So I put the source into my project for now, but I get this compilation error:

```
Predicate<Result<Double, Error>>' is not convertible to 'Predicate<Result<Double, CalculationError>>
```

...because `Result<Double, CalculationError>` is what I declared for that type in my program. The `Failure` type of `Result` doesn't necessarily have to conform to `Error`.

It works better for me if I let the `Failure` type of `Result` be parameterized as well, by changing this line to:

```
public func beSuccess<T, E>(test: ((T) -> Void)? = nil) -> Predicate<Result<T, E>> {
```